### PR TITLE
Async cluster

### PR DESCRIPTION
**Pull request by shivam-tripathi**
Add asyncCluster module. Read README.md for detailed explanation.

### CODE REVIEW

File: src/asyncCluster/README.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+5. **instanceFunction**: The base function to be run for each argument. It takes in arguments in form of an object with _base_ key as one of the _clusterArgs_, _id_ key as _workerId_ and _init_ key as _return value of init function_ ( as explained above).
+	```javascript
+	function instanceFunction({id:number, init: ?, base: ?}) {...}
+	```
+	* It is assumed that the base functions return a `Promise` carrying the results computed after processing on the _workerArgs_.
+	* In case of data import producers, _instanceFunction_ take in a file path, create a file IO stream and process the file.  They return the results in a `Promise`, which contains results fulfilled after completion of the processing. The worker process is terminated after fulfilment of the `Promise`.
+	* In case of data import consumers, _instanceFunction_ fires _#consume_ function, which is theoretically supposed to stay alive indefinitely (as long as the queue lives). As a result, the _instanceFunction_ returns a never fulfilling `Promise`, so that the _#consume_ function continues to listen for new arrivals. It is only terminated using something like `SIGINT` signal from keyboard (Ctrl+C).
```
[the remark this answers is not in the document]
Comment (shivam-tripathi) [reply src/asyncCluster/README.md]: I intended the module API to be generic, it ends the worker process when it receives results from the instance functions (like in producers). But the consumers' instance function should never return a value, and run indefinitely - so I couldn't think of any other way to keep the API consistent and still make the consumers run indefinitely.

**Comment by shivam-tripathi** [comment]
Made the changes, left some comments. Ready for review again :)

File: src/asyncCluster/async.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+function getAsyncWorkerFunctions({instanceFunction, allWorkerArgs}) {
+	function asyncWorkerFunction(args) {
+		return (callback) => callback(null, instanceFunction(args));
+	}
+
+	return allWorkerArgs.map(args => asyncWorkerFunction(args));
+}
+
+/**
+ * Extract the import configuration from the config file.
```
Comment (LordSputnik) [inline src/asyncCluster/async.js]: This description doesn't seem right.

File: src/asyncCluster/cluster.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+import * as Error from '../helpers/errors';
+import asyncWorker from './async';
+import cluster from 'cluster';
+import {isNotDefined} from '../helpers/utils';
+import log from '../helpers/logger';
+
+
+/**
+ * Extract the import configuration from the config file.
```
Comment (LordSputnik) [inline src/asyncCluster/cluster.js]: This description also doesn't seem right.

File: src/asyncCluster/cluster.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+ * 		chunks, split into number of workers
+ * @param {Object} args - Remaining arguments passed
+ */
+export default function runClusters({
+	masterExitCallback,
+	processLimit,
+	workerArgsArr,
+	...args
+}) {
+	if (typeof masterExitCallback !== 'function') {
```
Comment (LordSputnik) [inline src/asyncCluster/cluster.js]: Should use `_.isFunction` here

File: src/asyncCluster/async.js (hunk 2 above)
Comment (shivam-tripathi) [reply src/asyncCluster/async.js]: Whoops!

**Pull request merged**